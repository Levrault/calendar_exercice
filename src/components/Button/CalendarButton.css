.button {
  background-color: transparent;
  border: 0;
  border-radius: 50%;
  color: $palette-primary-text;
  cursor: pointer;
  display: flex;
  flex: $space-l 0 0;
  height: $space-l;
  justify-content: center;
  transition: background-color .25s;

  &.active,
  &:hover {
    background-color: $palette-light-primary;
  }
}
