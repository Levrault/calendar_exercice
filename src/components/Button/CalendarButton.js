import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './CalendarButton.css';

/**
 * Calendar button
 * @param {bool} [active=false]
 * @param {string} [color]
 * @param {string} [fontColor]
 * @param {node} children
 * @param {function} onClick
 * @returns {node}
 */
const CalendarButton = ({ active, color, fontColor, children, onClick }) => {
  const className = classnames(styles.button, {
    [styles.active]: active
  });
  const style = active ? { backgroundColor: color, color: fontColor } : {};
  return (
    <button className={className} style={style} onClick={onClick} >
      {children}
    </button>
  );
};

CalendarButton.propTypes = {
  active: PropTypes.bool,
  color: PropTypes.string,
  fontColor: PropTypes.string,
  onClick: PropTypes.func
};

CalendarButton.defaultProps = {
  active: false
};

export default CalendarButton;
