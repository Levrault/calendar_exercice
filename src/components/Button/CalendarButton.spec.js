import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from '../../tools/test/enzyme';
import CalendarButton from './CalendarButton';

describe('CalendarButton component', () => {
  it('should render without being active or in edit mode', () => {
    const component = renderer.create(
      <CalendarButton
        color="#EEEEEE">
        CalendarButton
      </CalendarButton>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with active mode on', () => {
    const component = renderer.create(
      <CalendarButton
        color="#EEEEEE"
        active>
        CalendarButton
      </CalendarButton>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call onClick function', () => {
    const onClick = jest.fn();
    const component = shallow(
      <CalendarButton
        onClick={onClick}
        color="#EEEEEE"
        active>
        CalendarButton
      </CalendarButton>,
    );

    component.simulate('click');
    expect(onClick).toHaveBeenCalled();
  });
});
