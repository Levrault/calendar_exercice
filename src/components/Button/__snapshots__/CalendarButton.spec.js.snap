// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarButton component should render with active mode on 1`] = `
<button
  className="undefined"
  style={
    Object {
      "backgroundColor": "#EEEEEE",
      "color": undefined,
    }
  }
>
  CalendarButton
</button>
`;

exports[`CalendarButton component should render without being active or in edit mode 1`] = `
<button
  className=""
  style={Object {}}
>
  CalendarButton
</button>
`;
